mlserver/codecs/utils.py

```diff
@@ -45,6 +60,45 @@ def _save_decoded(parametrised_obj: Parametrised, decoded_payload: Any):
     setattr(parametrised_obj.parameters, DecodedParameterName, decoded_payload)
 
 
+def encode_response_output(
+    payload: Any,
+    request_output: RequestOutput,
+    metadata_outputs: Dict[str, MetadataTensor] = {},
+) -> Optional[ResponseOutput]:
+    output_metadata = metadata_outputs.get(request_output.name)
+    content_type = _get_content_type(request_output, output_metadata)
+    codec = (
+        find_input_codec(content_type)
+        if content_type
+        else find_input_codec_by_payload(payload)
+    )
+
+    if not codec:
+        return None
+
+    return codec.encode(
+        name=request_output.name,
+        payload=payload,
+    )
+
+
+def encode_inference_response(
+    payload: Any,
+    model_settings: ModelSettings,
+) -> Optional[InferenceResponse]:
+    # TODO: Allow users to override codec through model's metadata
```

Comment:
It should already be possible to override the content type either via the `outputs` field within the **request payload** (you can "request outputs" and provide details about them), or through the model's metadata.

The only missing piece is doing the same thing for the response-level codecs. Right now, we only have a single `content_type` field at the request-level (both within the request payload and the model's metadata), but that's already used to decide the codec to use for the request payload, which could be different that the one required for the response. 

A potential workaround would be to add a `response_content_type` field, within the `Parameters` object, but I'm not too keen on that idea. It would be great to hear your thoughts on other options to tackle this one!

Reply:
ok got it